### DISCUSSION TITLE
why naive-ui component can get props onXXX in `setup` ?

### DISCUSSION THREAD

**Discussion by Dedicatus546** [tusen-ai/naive-ui#3706] [Q&A]
naive-ui

![image](https://user-images.githubusercontent.com/48575405/189563849-5fec02ae-bc9b-4406-8673-3cdad6782282.png)

my-project

![image](https://user-images.githubusercontent.com/48575405/189563903-5cf5a36f-8ea2-4886-8947-e3b0e84da38f.png)

it worried me a long time...

**Comment by 07akioni** [comment]
I don't know, you may check typescript version of those 2 projects.

**Reply by 07akioni** [reply]
I expected `onXxx` appears in `props` if I specified it.

**Reply by Dedicatus546** [reply]
i update dep's version and it work well. but when i recocery last version, the problem disappear... 

and i create a new project, it work well.

maybe dep's version cause. 

Thanks!
